Automattic/jetpack
# VideoPress: Add endpoint to serve featured stats for My Jetpack

Relates to #29753.

Blocks #29764.

## Proposed changes:

* Added `videopress/v1/stats/featured` endpoint to provide featured stats that will be used on My Jetpack
* The following data is being provided:
   * views for the last 7 days and the 7 days before it
   * impressions for the last 7 days and the 7 days before it
   * watch time for the last 7 days and the 7 days before it

This will be enough for the first iteration of the My Jetpack VideoPress card enhancement and will set the basic structure for adding more feature stats on the next iterations.

The idea of providing this from the VideoPress package is to keep the specific logic and calculations inside the package, and just to basic consumption of it from the My Jetpack frontend.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Make sure to test on a site that has some videos and that will have views or impressions; if you don't have one, just upload a couple of videos, add them to some post and play the videos some times
* Go to some page on the `wp-admin` of your site, open the browser console and run the following snippet:

```
wp.apiFetch({ path: "/videopress/v1/stats/featured" })
  .then((data) => console.log(data));
```

* This will do a request to the endpoint and print the data on the console
* The expected response is similar to this:

```
{
    "label": "last 7 days",
    "data": {
        "views": {
            "current": 4,
            "previous": 3
        },
        "impressions": {
            "current": 8,
            "previous": 11
        },
        "watch_time": {
            "current": 0.013888888888888888,
            "previous": 0.005555555555555556
        }
    }
}
```

The response contains the label for the kind of metric we are returning (may be useful on the frontend) as well as the value for the three basic metrics for the current and the previous period (7 days window).
